src/Lumina/GameData.cs

```diff
@@ -78,10 +79,27 @@ public GameData( string dataPath, LuminaOptions? options = null! )
                 throw new ArgumentException( "the data path arg must point to the sqpack directory", nameof( dataPath ) );
             }
 
-            Repositories = new Dictionary< string, Repository >();
-            foreach( var repo in DataPath.GetDirectories() )
+            if( options?.LoadMultithreaded == true )
             {
-                Repositories[ repo.Name.ToLowerInvariant() ] = new Repository( repo, this );
+                var repoTask = new Dictionary< string, Task< Repository > >();
+                foreach( var repo in DataPath.GetDirectories() )
```

Comment:
why not just use parallel.foreach?

Reply:
Parallel.ForEach does not provide task return values, so working around it would have been requiring more code. But I've changed again to effectively just wait once per function, and this did make loading quicker by 200ms. On the other hand, using AsParallel was slower by 400ms.

Reply:
yeah new stuff looks nicer in any case, should've had a better look at it before lol